### DISCUSSION TITLE
Avoid implementing TelegramLongPollingBot

### DISCUSSION THREAD

**Discussion by ggiambo** [rubenlagus/TelegramBots#1331] [Q&A] [answered]
For some reasons, I don't want to implement `TelegramLongPollingBot` but instead have an implementation for `DefaultAbsSender` and one for `LongPollingBot`.
Well, it seems that I'm not getting updates, `LongPollingBot#onUpdateReceived` ist not being called.

I'm doing something wrong, is it doable?

**Comment by Chase22** [comment]
Did you register the LongPollingBot with the API?

```java
TelegramBotsApi telegramBotsApi = new TelegramBotsApi(DefaultBotSession.class);
telegramBotsApi.registerBot(new YourBot());
```

**Comment by ggiambo** [comment]
> Did you register the LongPollingBot with the API?

Yes, I did it, directly into the implementation of `LongPollingBot`  (Kotlin, Springboot):

```kotlin
@Service
class MyAwesomeBot(...) : LongPollingBot {

    @PostConstruct
    fun postConstruct() {
        TelegramBotsApi(DefaultBotSession::class.java).registerBot(this)
    }
```

 I guess there's something "wrong" in what I'm doing.
I'll try to create a stripped-down, minimalist, version of the project, in order to reproduce the problem.

**Comment by ggiambo** [comment, marked as answer]
Looks  like I cannot reproduce it in the "minimalist" version of the project.
That means that is doable and that I'm making something wrong.
